packages/js-sdk/src/sandbox.ts

```diff
@@ -170,149 +264,253 @@ export class Sandbox extends SandboxBase {
+    );
   }
 
   /**
-   * Get the current screen size.
-   * @returns An object with `width` and `height` properties.
-   */
-  async getScreenSize() {
-    // We save the value to a file because stdout contains warnings about Xauthority.
-    await this.runPyautoguiCode(`
-width, height = pyautogui.size()
-with open("/tmp/size.txt", "w") as f:
-    f.write(str(width) + " " + str(height))
-`)
-    // Size is like this: 100 200
-    const size = await this.files.read('/tmp/size.txt')
-    const [width, height] = size.split(' ').map(Number)
-    return { width, height }
+   * Get the URL to connect to the VNC server.
+   * @param autoConnect - Whether to automatically connect to the server after opening the URL.
+   * @returns The URL to connect to the VNC server.
+   */
+  public getUrl(autoConnect: boolean = true): string {
+    let url = new URL(this.url);
+    if (autoConnect) {
+      url.searchParams.set('autoconnect', 'true');
+    }
+    if (this.desktop.novncAuthEnabled) {
+      url.searchParams.set("password", this.password);
+    }
+    return url.toString()
   }
 
   /**
-   * Run the given Python code that uses pyautogui.
+   * Start the VNC server.
    */
-  async runPyautoguiCode(
-    code: string,
-    opts: {
-      onStdout?: (data: string) => void
-      onStderr?: (data: string) => void
-    } = {},
-  ) {
-    const path = `/tmp/code-${generateRandomId()}.py`
-    const wrappedCode = this.wrapPyautoguiCode(code)
-    await this.files.write(path, wrappedCode)
-    const out = await this.commands.run(`python ${path}`, {
-      onStdout: opts.onStdout,
-      onStderr: opts.onStderr,
-    })
+  public async start(): Promise<void> {
+    await this.stop(); // If start is called while the server is already running, we just restart it
 
-    this.files.remove(path)
-    return out
-  }
+    if (this.vncCommand === "") {
+      await this.setVncCommand();
+    }
+    this.vncHandle = await this.desktop.commands.run(this.vncCommand, { background: true });
+    if (!await this.waitForPort(this.desktop.vncPort)) {
+      throw new Error("Could not start VNC server");
+    }
 
-  private wrapPyautoguiCode(code: string) {
-    return `
-import pyautogui
-import os
-import Xlib.display
+    this.novncHandle = await this.desktop.commands.run(this.novncCommand, { background: true });
+    if (!await this.waitForPort(this.desktop.streamPort)) {
+      throw new Error("Could not start noVNC server");
+    }
+  }
 
-display = Xlib.display.Display(os.environ["DISPLAY"])
-pyautogui._pyautogui_x11._display = display
+  /**
+   * Stop the VNC server.
+   */
+  public async stop(): Promise<void> {
+    if (this.vncHandle) {
+      await this.vncHandle.kill();
+      this.vncHandle = null;
+    }
 
-${code}
-exit(0)
-`
+    if (this.novncHandle) {
+      await this.novncHandle.kill();
+      this.novncHandle = null;
+    }
   }
 }
+
+export class Sandbox extends Desktop { }
```

Comment:
if you're exporting the Sandbox here it should be e2b Sandbox class, not Desktop